forte/processors/data_augment/algorithms/embedding_similarity_augmenter.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+from typing import Dict, List
+import numpy as np
+from tensorflow import gfile
+from texar.tf.data import Embedding, load_glove
+from forte.processors.data_augment.algorithms.base_augmenter import BaseDataAugmenter
+
+__all__ = [
+    "EmbeddingSimilarityAugmenter",
+]
+
+random.seed(0)
+
+emb_types = ['glove']
+
+
+def load_glove_vocab(filename):
```

Comment:
It is slightly ad-hoc to implement embeddng vocab here, let's do it formally.

You can use the Texar embedder class to wrap these pre-trained models: https://github.com/asyml/texar-pytorch/blob/master/texar/torch/modules/embedders/embedders.py

And you can follow the examples here to allow automatically download the Glove vector if possible: https://github.com/asyml/texar-pytorch/blob/master/texar/torch/modules/pretrained/pretrained_base.py#L39

Reply:
Thanks for the comment. However, I am a bit confused about what to do here:

I didn't find a method in the `Embedder` class that can take a word embedding file to initialize the embeddings. 

However, there are methods in the `texar.tf.data.Embedding` class that can do this, which is why I chose to use that class.

Also, it seems like these two aren't connected: I didn't find a way to use an `Embedding` to initialize `Embedder`, or vice versa. Also, `Embedding` uses Numpy array, and `Embedder` uses Pytorch tensor.

I think what I could do is: initialize `Embedding` first, then use that to initialize the `Embedder`.  However, I don't see the value of this additional wrapper because we will need to use the entire embedding matrix anyway.

Reply:
I see, I just realize you already using Texar Embedding. But what I mean here is we should do it outside the class so user can pass in his own embedding class. For word embedding I agree that we should use the Texar Embedding class, Embedder can be used for contextual embeddings.